nixos/modules/services/misc/ringboard.nix

```diff
@@ -0,0 +1,64 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.ringboard;
+in
+{
+
+  options.services.ringboard = {
+    enable = lib.mkEnableOption "Ringboard clipboard manager";
+
+    package = lib.mkPackageOption pkgs "ringboard" { };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.user.services.ringboard-server = {
+      description = "Ringboard server";
+      documentation = [ "https://github.com/SUPERCILEX/clipboard-history" ];
+      after = [ "multi-user.target" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        Type = "notify";
+        Environment = "RUST_LOG=trace";
+        ExecStart = "${cfg.package}/bin/ringboard-server";
+        Restart = "on-failure";
+        Slice = "ringboard.slice";
+      };
+    };
+
+    systemd.user.services.ringboard-listener = {
+      description = "Ringboard clipboard listener (auto: X11/Wayland)";
+      documentation = [ "https://github.com/SUPERCILEX/clipboard-history" ];
+      requires = [ "ringboard-server.service" ];
+      after = [
+        "ringboard-server.service"
+        "graphical-session.target"
+      ];
+      bindsTo = [ "graphical-session.target" ];
+      wantedBy = [ "graphical-session.target" ];
+      script = ''
+        if [ "$XDG_SESSION_TYPE" = "wayland" ]; then
+          exec '${cfg.package}'/bin/ringboard-wayland
+        else
+          exec '${cfg.package}'/bin/ringboard-x11
+        fi
+      '';
+      environment.RUST_LOG = "trace";
+      serviceConfig = {
+        Type = "exec";
+        Restart = "on-failure";
+        Slice = "ringboard.slice";
+      };
+    };
+
+    systemd.user.slices.ringboard = {
```

Comment:
It's probably not a big deal, but maybe this should be a subslice of `session.slice`? (see https://www.freedesktop.org/software/systemd/man/latest/systemd.special.html#Special%20User%20Slice%20Units). I suppose it would have to be renamed `slices.session-ringboard`.

I see upstream uses `ringboard.slice`, so either one is probably fine, feel free to just mark as resolved if you're not interested.

Reply:
Sure. The idea is that if you just rename the slice from `ringboard` to `session-ringboard`, it will automatically become a subslice of `session.slice` (you can verify this with `systemctl --user status`)

This will make the slice and its units inherit the properties of `session.slice`, which is meant for services needed for the graphical session to work properly. I'm not sure whether it's actually configured differently, or whether it just *allows* you to configure the slice differently, but according to the docs `session.slice` should be optimized for rapid scheduling and non-intensive work, making the service more responsive.